src/indigenousLayers.js

```diff
@@ -82,10 +82,10 @@ L.LayerGroup.IndigenousLayers = L.LayerGroup.extend(
               var ill_poly ;
```

Comment:
@rexagod , how can we possibly use the ternary operator here (in the second if else block)?
All 3 assignments to ill_poly are very different ie not only is the name different (hence this.name based assignment wont work) All three of them are linked to a separate link here.(href part)

Reply:
```js
function getPoly(coords, clr, ...stringparams) {
  return L.polygon(coords, {color: clr}).bindPopup(
    `<strong>Name : </strong> ${stringparams.nme} <br>
    <strong>Description: </strong> <a href=${stringparams.desc}>Native Lands - ${stringparams.nme}</a><br>
    <i>From the <a href='https://github.com/publiclab/leaflet-environmental-layers/pull/${
      this.name === "Territories" ? 77 : this.name === "Languages" ? 76 : 78
    }'>Indigenous ${this.name} Inventory</a> (<a href='https://publiclab.org/notes/sagarpreet/06-06-2018/leaflet-environmental-layer-library?_=1528283515'>info<a>)</i>`
  );
}

var ill_poly = getPoly(...options);
```

Reply:
Hi @rexagod, Thanks for the review!
I tried this code. It initially gave me a reference to option not defined error on my console. Then I removed options passed coords and clr and now it works but the Language layer is loading for both territories and Treties layer as well unlike before with the if else block.

Reply:
```js
// assuming that `this.name` is defined in this scope

function getPoly(coords, clr, ...stringparams) {
  return L.polygon(coords, {color: clr}).bindPopup(
    `<strong>Name : </strong> ${stringparams[0]} <br>
    <strong>Description: </strong> <a href=${stringparams[1]}>Native Lands - ${stringparams[0]}</a><br>
    <i>From the <a href='https://github.com/publiclab/leaflet-environmental-layers/pull/${
      this.name === "Territories" ? 77 : this.name === "Languages" ? 76 : 78
    }'>Indigenous ${this.name} Inventory</a> (<a href='https://publiclab.org/notes/sagarpreet/06-06-2018/leaflet-environmental-layer-library?_=1528283515'>info<a>)</i>`
  );
}

// options is an array
var options = [coords, 'blue', 'mario', "mario's desc"];

var ill_poly = (isNaN(coords[0][0][0]) || isNaN(coords[0][0][1])) ? undefined : getPoly(...options);
```

@ananyaarun You need to pass an array containing the required params needed to initialize the `getPoly` method. The code above was only for reference, so you can copy the logic used, not the code itself, but if you need the exact code, you can use the snippet above (with obvious appropriate changes).

![Untitled](https://user-images.githubusercontent.com/33557095/59861064-c2de2e00-939d-11e9-8d7d-f596bf7b604d.png)

PS. I replaced the "L.polygon" and "bindPopup" methods with simple object notations since they weren't defined in the scope, but should work well once you place this inside your js file with all the appropriate scopes defined. Also, one can infer from the image above that the ternary seems to work fine, I hope you get the idea now.

Reply:
```
function getPoly(coords, clr, ...stringparams) {
  return L.polygon(coords, {color: clr}).bindPopup(
    `<strong>Name : </strong> ${stringparams[0]} <br>
     <strong>Description: </strong> <a href=${stringparams[1]}>Native Lands - ${stringparams[0]}           <a><br><i>From the <a href='https://github.com/publiclab/leaflet-environmental-layers/pull/${
       this.name === "Territories" ? 77 : this.name === "Languages" ? 76 : 78
   }'>Indigenous ${this.name} Inventory</a> (<a href='https://publiclab.org/notes/sagarpreet/06-06-2018/leaflet-environmental-layer-library?_=1528283515'>info<a>)</i>`
  );
}
var array = [coords, clr, 'nme', "desc"];
var ill_poly = (isNaN(coords[0][0][0]) || isNaN(coords[0][0][1])) ? undefined : getPoly(...array);
```

So here is my console, I see that despite choosing the territories layer, the link has 78 (instead of 77). yes the ternary is working but this.name is not accessible here.
I see that it appears as blank on the link too.

I am trying to see what is going wrong 
Thanks!

Reply:
@ananyaarun `this` is actually `window` in the console, so `this.name` translates to `window.name` **in the console**. The snippet above will work, as I stated, *once you place this inside your js file with all the appropriate scopes defined*, but if you'd like to test it out in your console, you can `this.name="Territorries"` (inside the console) or `window.name = "Territories"` (inside your js file).